lib/src/git_backend.rs

```diff
@@ -1540,67 +1539,100 @@ mod tests {
     fn read_plain_git_commit(uses_tree_conflict_format: bool) {
         let settings = user_settings();
         let temp_dir = testutils::new_temp_dir();
         let store_path = temp_dir.path();
         let git_repo_path = temp_dir.path().join("git");
-        let git_repo = git2::Repository::init(git_repo_path).unwrap();
+        let git_repo = git_init(git_repo_path);
 
         // Add a commit with some files in
-        let blob1 = git_repo.blob(b"content1").unwrap();
-        let blob2 = git_repo.blob(b"normal").unwrap();
-        let mut dir_tree_builder = git_repo.treebuilder(None).unwrap();
-        dir_tree_builder.insert("normal", blob1, 0o100644).unwrap();
-        dir_tree_builder.insert("symlink", blob2, 0o120000).unwrap();
-        let dir_tree_id = dir_tree_builder.write().unwrap();
-        let mut root_tree_builder = git_repo.treebuilder(None).unwrap();
+        let blob1 = git_repo.write_blob(b"content1").unwrap().detach();
+        let blob2 = git_repo.write_blob(b"normal").unwrap().detach();
+        let mut dir_tree_editor = git_repo.empty_tree().edit().unwrap();
+        dir_tree_editor
+            .upsert("normal", gix::object::tree::EntryKind::Blob, blob1)
+            .unwrap();
+        dir_tree_editor
+            .upsert("symlink", gix::object::tree::EntryKind::Link, blob2)
+            .unwrap();
+        let dir_tree_id = dir_tree_editor.write().unwrap().detach();
+        let mut root_tree_builder = git_repo.empty_tree().edit().unwrap();
         root_tree_builder
-            .insert("dir", dir_tree_id, 0o040000)
+            .upsert("dir", gix::object::tree::EntryKind::Tree, dir_tree_id)
             .unwrap();
-        let root_tree_id = root_tree_builder.write().unwrap();
-        let git_author = git2::Signature::new(
-            "git author",
-            "git.author@example.com",
-            &git2::Time::new(1000, 60),
-        )
-        .unwrap();
-        let git_committer = git2::Signature::new(
-            "git committer",
-            "git.committer@example.com",
-            &git2::Time::new(2000, -480),
-        )
-        .unwrap();
-        let git_tree = git_repo.find_tree(root_tree_id).unwrap();
+        let root_tree_id = root_tree_builder.write().unwrap().detach();
+        let git_author = gix::actor::Signature {
+            name: bstr::BString::from("git author"),
+            email: bstr::BString::from("git.author@example.org"),
+            time: gix::date::Time::new(1000, 60 * 60),
+        };
+        let git_committer = gix::actor::Signature {
+            name: bstr::BString::from("git committer"),
+            email: bstr::BString::from("git.committer@example.org"),
+            time: gix::date::Time::new(2000, -480 * 60),
+        };
         let git_commit_id = git_repo
-            .commit(
-                None,
-                &git_author,
+            .commit_as(
                 &git_committer,
+                &git_author,
+                "refs/heads/dummy",
```

Comment:
nit: maybe better to delete the dummy ref because we're testing core `GitBackend` functionality.